packages/shared/lib/lock/locking.ts

```diff
@@ -0,0 +1,43 @@
+import type { KVStore } from '../kvstore/KVStore.js';
+
+export class Locking {
+    private store: KVStore;
+
+    constructor(store: KVStore) {
+        this.store = store;
+    }
+
+    public async tryAcquire(key: string, ttlInMs: number, acquitistionTimeoutMs: number): Promise<void> {
+        if (ttlInMs <= 0) {
+            throw new Error(`lock's TTL must be greater than 0`);
+        }
+        if (acquitistionTimeoutMs <= 0) {
+            throw new Error(`acquitistionTimeoutMs must be greater than 0`);
```

Comment:
How is this error handled? What would be surfaced to the user if this happens?

In other places in the codebase a `ServiceResponse` is used in conjunction with a friendly error message via `NangoError` to let the user know there was a problem

Reply:
The error is handled by the caller which is in this case `connection.service`, which itself return the NangoError https://github.com/NangoHQ/nango/blob/ae746bef77398592cda5c5fa28b28d2e300f831e/packages/shared/lib/services/connection.service.ts#L620

Reply:
Ok, great, thanks